### PR TITLE
8304912: Use OperatingSystem enum in java.desktop module

### PR DESCRIPTION
**Pull request by RogerRiggs**
Update classes in the java.desktop module to use the jdk.internal.util.OperatingSystem enum instead of the `os.name` system property to select OS specific behaviors.

---------
### Progress
- [x] Change must be properly reviewed (1 review required, with at least 1 [Reviewer](https://openjdk.org/bylaws#reviewer))
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue

### Issue
 * [JDK-8304912](https://bugs.openjdk.org/browse/JDK-8304912): Use OperatingSystem enum in java.desktop module

### Reviewers
 * [Brian Burkhalter](https://openjdk.org/census#bpb) (@bplb - **Reviewer**) ⚠️ Review applies to [7f93ecbf](https://git.openjdk.org/jdk/pull/13213/files/7f93ecbffa1b44827ad5228a417d62a5b6247b1d)
 * [Phil Race](https://openjdk.org/census#prr) (@prrace - **Reviewer**) ⚠️ Review applies to [a3334e88](https://git.openjdk.org/jdk/pull/13213/files/a3334e88440eb576943e20daa7d5ee8f824d1290)
 * [Andrey Turbanov](https://openjdk.org/census#aturbanov) (@turbanoff - Committer) ⚠️ Review applies to [a3334e88](https://git.openjdk.org/jdk/pull/13213/files/a3334e88440eb576943e20daa7d5ee8f824d1290)
 * @Glavo (no known openjdk.org user name / role)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.org/jdk.git pull/13213/head:pull/13213` \
`$ git checkout pull/13213`

Update a local copy of the PR: \
`$ git checkout pull/13213` \
`$ git pull https://git.openjdk.org/jdk.git pull/13213/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 13213`

View PR using the GUI difftool: \
`$ git pr show -t 13213`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.org/jdk/pull/13213.diff">https://git.openjdk.org/jdk/pull/13213.diff</a>

</details>

### Webrev
[Link to Webrev Comment](https://git.openjdk.org/jdk/pull/13213#issuecomment-1487388968)

### CODE REVIEW

**Comment by turbanoff** [comment]
sun.awt.OSInfo could be cleanup up:
1. sun.awt.OSInfo#getOSTypeAction - is unused now
2. sun.awt.OSInfo#getOSType could use `OperatingSystem` method instead of relying of sun.awt.OSInfo#OS_NAME property

**Comment by RogerRiggs** [comment]
> * sun.awt.OSInfo#getOSTypeAction - is unused now
> 
> * sun.awt.OSInfo#getOSType could use `OperatingSystem` method instead of relying of sun.awt.OSInfo#OS_NAME property. It seems it's used only in one place - sun.awt.X11.MotifColorUtilities#loadSystemColorsForCDE

OSInfo.getOSType() is used by some tests; those references will need to be remediated.

**Comment by turbanoff** [comment]
Let's at least update `sun.awt.X11.MotifColorUtilities#loadSystemColorsForCDE`
https://github.com/openjdk/jdk/blob/master/src/java.desktop/unix/classes/sun/awt/X11/MotifColorUtilities.java#L440

**Review by bplb — APPROVED** [review]
Looks fine modulo any cleanup to be performed subsequently.

**Comment by RogerRiggs** [comment]
> Looks fine modulo any cleanup to be performed subsequently.

Tracked with [JDK-8305606](https://bugs.openjdk.org/browse/JDK-8305606)

File: src/java.desktop/share/classes/sun/font/FontUtilities.java [hunk 1]
```diff
@@ -64,11 +65,10 @@ private static void initStatic() {
             @SuppressWarnings("deprecation") // PlatformLogger.setLevel is deprecated.
             @Override
             public Object run() {
-                String osName = System.getProperty("os.name", "unknownOS");
 
-                isLinux = osName.startsWith("Linux");
+                isLinux = OperatingSystem.isLinux();
 
-                isMacOSX = osName.contains("OS X"); // TODO: MacOSX
+                isMacOSX = OperatingSystem.isMacOS(); // TODO: MacOSX
```
Comment (turbanoff) [inline src/java.desktop/share/classes/sun/font/FontUtilities.java:71]: I think this `TODO` is not actual anymore

**Comment by turbanoff** [comment]
Found another couple of places:

https://github.com/openjdk/jdk/blob/7cf24d1c06142a3bab9cce5cd0ba34b8bbccf00f/src/java.desktop/unix/classes/sun/awt/X11/MotifColorUtilities.java#L641
https://github.com/openjdk/jdk/blob/7cf24d1c06142a3bab9cce5cd0ba34b8bbccf00f/src/java.desktop/unix/classes/sun/awt/X11/XScrollbarPeer.java#L78

**Comment by prrace** [comment]
So if we want to get rid of all the getProperty (etc) calls and unify it then  the way I'd prefer to do this is via a single channel inside the desktop module.
So change everything to use that pre-existing sun.awt.OSInfo class.
Modifying that as necessary. Then that class can be the SINGLE place that has a direct dependency on this java.base internal and we can more easily drop it if required.
Meanwhile I also think there's a better than 50:50 chance someone will add new uses of System.getProperty .. whatever the solution is. So whack-a-mole-away.

File: src/java.desktop/share/classes/sun/awt/shell/ShellFolder.java [hunk 2]
```diff
@@ -1,5 +1,5 @@
 /*
- * Copyright (c) 2000, 2022, Oracle and/or its affiliates. All rights reserved.
+ * Copyright (c) 2000, 2023, Oracle and/or its affiliates. All rights reserved.
```
Comment (prrace) [inline src/java.desktop/share/classes/sun/awt/shell/ShellFolder.java:2]: no changes so revert (c) year

File: src/java.desktop/share/classes/sun/swing/WindowsPlacesBar.java [hunk 3]
```diff
@@ -1,5 +1,5 @@
 /*
- * Copyright (c) 2003, 2018, Oracle and/or its affiliates. All rights reserved.
+ * Copyright (c) 2003, 2023, Oracle and/or its affiliates. All rights reserved.
```
Comment (prrace) [inline src/java.desktop/share/classes/sun/swing/WindowsPlacesBar.java:2]: no changes so revert (c) year

File: src/java.desktop/unix/classes/sun/font/FcFontConfiguration.java [hunk 4]
```diff
@@ -1,5 +1,5 @@
 /*
- * Copyright (c) 2008, 2022, Oracle and/or its affiliates. All rights reserved.
+ * Copyright (c) 2008, 2023, Oracle and/or its affiliates. All rights reserved.
```
Comment (prrace) [inline src/java.desktop/unix/classes/sun/font/FcFontConfiguration.java:2]: no changes so revert (c) year

File: src/java.desktop/windows/classes/com/sun/java/swing/plaf/windows/WindowsLookAndFeel.java [hunk 5]
```diff
@@ -1,5 +1,5 @@
 /*
- * Copyright (c) 1997, 2021, Oracle and/or its affiliates. All rights reserved.
+ * Copyright (c) 1997, 2023, Oracle and/or its affiliates. All rights reserved.
```
Comment (prrace) [inline src/java.desktop/windows/classes/com/sun/java/swing/plaf/windows/WindowsLookAndFeel.java:2]: no changes so revert (c) year

**Review by prrace — CHANGES_REQUESTED** [review]

File: src/java.desktop/share/classes/sun/awt/OSInfo.java
Comment (turbanoff) [inline src/java.desktop/share/classes/sun/awt/OSInfo.java:1]: `sun.awt.OSInfo#OS_NAME` is now unused

File: src/java.desktop/share/classes/sun/awt/OSInfo.java [hunk 6]
```diff
@@ -58,11 +61,14 @@ and so the method getWindowsVersion() will return the constant for known OS.
     public static final WindowsVersion WINDOWS_VISTA = new WindowsVersion(6, 0);
     public static final WindowsVersion WINDOWS_7 = new WindowsVersion(6, 1);
```
Comment (prrace) [inline src/java.desktop/share/classes/sun/awt/OSInfo.java:62]: The  windows version map looks like it hasn't had any love in a long time.
Updating it is VERY out of scope for this fix as it needs some thought in whether to remove old entries
and what to do about replacing them so I filed https://bugs.openjdk.org/browse/JDK-8305777

File: src/java.desktop/windows/classes/com/sun/java/swing/plaf/windows/WindowsLookAndFeel.java (hunk 5 above)
Comment (turbanoff) [reply src/java.desktop/windows/classes/com/sun/java/swing/plaf/windows/WindowsLookAndFeel.java:2]: @RogerRiggs missed this one

File: src/java.desktop/share/classes/sun/awt/OSInfo.java [hunk 7]
```diff
@@ -76,7 +82,7 @@ and so the method getWindowsVersion() will return the constant for known OS.
 
     private static final PrivilegedAction<OSType> osTypeAction = new PrivilegedAction<OSType>() {
```
Comment (turbanoff) [inline src/java.desktop/share/classes/sun/awt/OSInfo.java:83]: As this field is not used in production code anymore, I think it makes sense to inline it (allocate directly in `sun.awt.OSInfo#getOSTypeAction`)

**Review by turbanoff — APPROVED** [review]

**Comment by RogerRiggs** [comment]
@prrace Please review

**Review by prrace — APPROVED** [review]

File: src/java.desktop/share/classes/sun/awt/OSInfo.java [hunk 8]
```diff
@@ -25,6 +25,9 @@
 
 package sun.awt;
 
+import jdk.internal.util.OperatingSystem;
+
+import java.security.AccessController;
 import java.security.PrivilegedAction;
```
Comment (Glavo) [inline src/java.desktop/share/classes/sun/awt/OSInfo.java:31]: These imports are no longer used.

**Review by Glavo — APPROVED** [review]
It looks great.

**Pull request closed without merging**